src/widget/button.rs

```diff
@@ -55,16 +55,18 @@ impl Label {
         Label { text: text.into() }
     }
 
-    fn get_layout(&self, rt: &mut Piet, font_size: f64) -> <Piet as RenderContext>::TextLayout {
+    fn get_layout(
+        &self,
+        rt: &mut <Piet as RenderContext>::Text,
+        font_size: f64,
+    ) -> <Piet as RenderContext>::TextLayout {
         // TODO: caching of both the format and the layout
```

Comment:
If you're in here (and you feel like it) it might be a good time to resolve this TODO; we should only to construct the format once per font/size combo, and we should only need to construct the layout when the string changes.

Reply:
Well, honestly, I am not sure how exactly this should be implemented. 
>once per font/size combo

Do you mean using some hash map (per label or globally) to store formats or just constructing it only if font/size changes?

Reply:
What I mean is: we can store the font as an `Option<PietFont>`, in the label, and then only make it if it's missing; and we can store the layout as an `Option<(String, PietTextLayout)>`, and only make it when it's missing, or the string has changed.

Reply:
What I mean is: we can store the font as an `Option<PietFont>`, in the label, and then only make it if it's missing; and we can store the layout as an `Option<(String, PietTextLayout)>`, and only make it when it's missing, or the string has changed.

Reply:
@andlr are you interested in making this change now, or would you like me to merge this, and then you can do a follow-up (or not) later if you like?

Reply:
Yes, I think it is better to merge this now and add caching later with a separate PR